src/Confluent.Kafka/Impl/SafeKafkaHandle.cs

```diff
@@ -186,6 +187,19 @@ internal ErrorCode Produce(string topic, byte[] val, int valOffset, int valLengt
             var gchValue = default(GCHandle);
             var gchKey = default(GCHandle);
 
+            var matcher = new Regex("/([^:]+):(.*)");
+            var full_topic = topic;
+            if (matcher.Match(topic) == null)
```

Comment:
you're not concerned of the perf impact of doing a regex match on every produce?